src/Annotations/Command/QueueWatchCommand.php

```diff
@@ -66,8 +67,14 @@ protected function process(InputInterface $input, QueueItem $item, $entity = nul
                 $sanitized_display_name = $display_name;
             }
             $user = new User($id, $email, $sanitized_display_name);
-            $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
-            $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            try {
+                $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
+                $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            } catch (BadResponse $e) {
+                // If upsert failures then log error but don't repeat.
+                $this->queue->commit($item);
+                $this->logger->error(sprintf('Hypothesis user "%s" upsert failure.', $user->getUsername()));
```

Comment:
either log `$e` here, or rethrow it after this line to allow logging from the parent class

Reply:
Correcting myself: do not rethrow because the parent class will attempt to release `$item`. Log `$e` and then this code is done with it

Reply:
IIRC the logging method accepts the `$context` 2nd argument as `['exception' => $e]`